feature-account-impl/src/main/java/io/novafoundation/nova/feature_account_impl/presentation/manualBackup/accounts/ManualBackupSelectAccountViewModel.kt

```diff
@@ -0,0 +1,81 @@
+package io.novafoundation.nova.feature_account_impl.presentation.manualBackup.accounts
+
+import io.novafoundation.nova.common.base.BaseViewModel
+import io.novafoundation.nova.common.resources.ResourceManager
+import io.novafoundation.nova.common.utils.images.asIcon
+import io.novafoundation.nova.feature_account_api.presenatation.account.wallet.WalletUiUseCase
+import io.novafoundation.nova.feature_account_impl.R
+import io.novafoundation.nova.feature_account_impl.domain.manualBackup.ManualBackupInteractor
+import io.novafoundation.nova.feature_account_impl.domain.manualBackup.MetaAccountChains
+import io.novafoundation.nova.feature_account_impl.presentation.AccountRouter
+import io.novafoundation.nova.feature_account_impl.presentation.manualBackup.accounts.adapter.ManualBackupAccountGroupRVItem
+import io.novafoundation.nova.feature_account_impl.presentation.manualBackup.accounts.adapter.ManualBackupAccountRVItem
+import io.novafoundation.nova.feature_account_impl.presentation.manualBackup.accounts.adapter.ManualBackupRVItem
+import kotlinx.coroutines.flow.map
+
+class ManualBackupSelectAccountViewModel(
+    private val router: AccountRouter,
+    private val resourceManager: ResourceManager,
+    private val manualBackupInteractor: ManualBackupInteractor,
+    private val walletUiUseCase: WalletUiUseCase,
+    private val payload: ManualBackupSelectAccountPayload
+) : BaseViewModel() {
+
+    private val metaAccountChains = manualBackupInteractor.sortedMetaAccountChains(payload.metaId)
+        .shareInBackground()
+
+    val walletModel = metaAccountChains.map { metaAccountChains ->
+        walletUiUseCase.walletUiFor(metaAccountChains.metaAccount)
+    }
+
+    val accountsList = metaAccountChains.map { metaAccountChains ->
+        mapMetaAccountToUI(metaAccountChains)
+    }
+
+    fun walletClicked(accountModel: ManualBackupAccountRVItem) {
+        showError("Not implemented")
+    }
+
+    fun backClicked() {
+        router.back()
+    }
+
+    private fun mapMetaAccountToUI(
+        metaAccountChains: MetaAccountChains
+    ): List<ManualBackupRVItem> {
+        return buildList {
+            if (metaAccountChains.defaultChains.isNotEmpty()) {
```

Comment:
Iirc we have similar logic in Wallet Connect Session Approve screen. Can you please check whether we can reuse it or not?

Reply:
I checked it and think this logic is not very reusable in our case and also my logic is pretty simple and I think we can left it like that